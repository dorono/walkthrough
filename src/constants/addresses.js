export const ADDRESSES = {
    PEGNET_BURN_ADDRESS: 'EC2BURNFCT2PEGNETooo1oooo1oooo1oooo1oooo1oooo19wthin',
    IS_PRIVATE_KEY: 'IS_PRIVATE_KEY',
    IS_KEY: 'IS_KEY',
    IS_PROBABLY_A_KEY: 'IS_PROBABLY_A_KEY',
    IS_NOT_A_PRIVATE_ADDRESS: 'IS_NOT_A_PRIVATE_ADDRESS',
};

Object.freeze(ADDRESSES);
