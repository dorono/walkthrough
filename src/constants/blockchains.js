export const AVAILABLE_BLOCKCHAINS = {
    PUBLIC: {
        label: CONFIG.blockchainNetwork,
        value: 0,
        serviceName: 'mainnet',
        url: CONFIG.gatewayUrls.publicNet,
    },
    SHARED: {
        label: 'Shared Sandbox',
        value: 1,
        serviceName: 'shared',
        url: CONFIG.gatewayUrls.sharedSandbox,
    },
    PRIVATE: {
        label: 'Private Network',
        value: 2,
        serviceName: 'private_network',
        url: null,
    },
};

Object.freeze(AVAILABLE_BLOCKCHAINS);
