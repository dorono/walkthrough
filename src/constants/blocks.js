export const BLOCKS = {
    DIRECTORY: 'dblock', // Directory Block
    FACTOID: 'fblock', // Factoid Block
    ENTRY: 'eblock', // Entry Block
    PUBLIC_NETWORK: 'fnblock', // Public Factom Network
    BITCOIN_NETWORK: 'bnblock', // Cripto Block (Bitcoin and Ethereum)
    ETHEREUM_NETWORK: 'enblock', // Cripto Block (Bitcoin and Ethereum)
    DEFAULT: 'block', // Default Block
};

Object.freeze(BLOCKS);
