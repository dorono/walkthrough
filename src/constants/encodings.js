export const ENCODINGS = {
    FORMAT: {
        HEX: 'hex',
        RAW: 'raw',
        BASE64: 'base64',
        JSON: 'json',
    },
};

Object.freeze(ENCODINGS);
