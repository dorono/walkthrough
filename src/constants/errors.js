export const ERRORS = {
    CREDENTIALS: 'credentials',
    CUSTOM_CREDENTIALS_NOT_ALLOWED: 'custom-credentials-not-allowed',
    NETWORK: 'network',
    OTHER: 'other',
};

export const ERROR_MESSAGES = {
    AUTHENTICATION_FAILED: 'Authentication failed',
};

Object.freeze(ERRORS);
