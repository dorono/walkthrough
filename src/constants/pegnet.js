// these array items must be kept in the same order
export const PEGNET_TRANSACTION_TYPES = [
    'transfer',
    'conversion',
    'coinbase',
    'burn',
];
