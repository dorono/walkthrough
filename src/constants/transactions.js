export const TRANSACTIONS = {
    TYPE: {
        TRANSFER: {
            NUMBER: 1,
            NAME: 'Transfer',
        },
        CONVERSION: {
            NUMBER: 2,
            NAME: 'Conversion',
        },
        COINBASE: {
            NUMBER: 3,
            NAME: 'Coinbase',
        },
        BURN: {
            NUMBER: 4,
            NAME: 'Burn',
        },
    },
    TITLE: {
        INPUTS: 'Inputs',
        OUTPUTS: 'Outputs',
    },
    STATUSES: {
        PENDING: {
            LABEL: 'Pending',
            NUM_EXECUTED: 0,
            ALERT_TEXT: 'This transaction has not yet been processed by PegNet.',
            MESSAGE: 'Not yet available',
            TOOLTIP: 'Output amount is calculated based on conversion rate at time of processing',
        },
        REJECTED: {
            LABEL: 'Rejected',
            NUM_EXECUTED: -1,
            ALERT_TEXT: 'This transaction has been rejected due to insufficient funds.',
            MESSAGE: '(N/A)',
            TOOLTIP: 'This transaction has been rejected due to insufficient funds',
        },
    },
    PEGNET_PARENT_ROUTES: {
        TRANSACTIONS: 'transactions',
        ADDRESSES: 'addresses',
    },
    PEGNET_COMPLETED: 'pegnetCompleted',
    PEGNET_RECORDED: 'pegnetRecorded',
    PEGNET_CHAIN_ID: 'cffce0f409ebba4ed236d49d89c70e4bd1f1367d86402a3363366683265a242d',
    PEGNET_ASSET_LABELS: {
        PXAU: 'pGOLD',
        PXAG: 'pSILVER',
        PXBT: 'pBTC',
        PXBC: 'pBCH',
    },
    PARTIAL_CONVERSION_DIFFERENCE_LABEL: '(returned)',
    FCT_CONVERSION: 100000000,
};

Object.freeze(TRANSACTIONS);
