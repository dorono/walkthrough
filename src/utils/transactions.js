import {TRANSACTIONS} from 'constants/transactions';

export const isTransfer = transaction =>
    transaction.txaction === 1 && Array.isArray(transaction.outputs);

export const getPegnetTransactionName = txaction => {
    if (txaction === TRANSACTIONS.TYPE.TRANSFER.NUMBER) {
        return TRANSACTIONS.TYPE.TRANSFER.NAME;
    } else if (txaction === TRANSACTIONS.TYPE.COINBASE.NUMBER) {
        return TRANSACTIONS.TYPE.COINBASE.NAME;
    } else if (txaction === TRANSACTIONS.TYPE.CONVERSION.NUMBER) {
        return TRANSACTIONS.TYPE.CONVERSION.NAME;
    }

    return TRANSACTIONS.TYPE.BURN.NAME;
};

export const getTransactionStatus = transactionData => {
    if (transactionData.executed === TRANSACTIONS.STATUSES.PENDING.NUM_EXECUTED) {
        return TRANSACTIONS.STATUSES.PENDING.LABEL;
    } else if (transactionData.executed < TRANSACTIONS.STATUSES.PENDING.NUM_EXECUTED) {
        return TRANSACTIONS.STATUSES.REJECTED.LABEL;
    }

    return null;
};

export const generateTransactionList = (title, transactionData) => {
    // convert that data into an array for looping purposes
    let transactions = [transactionData];

    // for transfers, make sure that the list of outputs come from
    // the 'outputs' property of the transaction response
    if (title === TRANSACTIONS.TITLE.OUTPUTS) {
        if (isTransfer(transactions[0])) {
            transactions = transactions[0].outputs.map(output => {
                return {
                    user_address: output.address,
                    amount: output.amount,
                    unit: transactionData.fromasset,
                };
            });
        } else {
            transactions = [
                {
                    user_address: transactions[0].toaddress || transactions[0].fromaddress,
                    amount: transactions[0].toamount,
                    unit: transactions[0].toasset,
                },
            ];
        }
    } else {
        transactions = [
            {
                user_address: transactions[0].fromaddress,
                amount: transactions[0].fromamount,
                unit: transactions[0].fromasset,
            },
        ];
    }

    return transactions;
};

export const getOutputAmount = transaction => {
    if (isTransfer(transaction)) {
        return transaction.outputs.reduce(
            (accumulator, outputAmt) => accumulator + outputAmt.amount,
            0,
        );
    }

    return transaction.toamount;
};

export const getPropertyLabel = propertyName => {
    if (propertyName && TRANSACTIONS.PEGNET_ASSET_LABELS[propertyName.toUpperCase()]) {
        return TRANSACTIONS.PEGNET_ASSET_LABELS[propertyName.toUpperCase()];
    }

    return propertyName;
};

export const getPegnetLabel = propertyName => {
    if (Object.values(TRANSACTIONS.PEGNET_ASSET_LABELS).indexOf(propertyName) >= 0) {
        const key = Object.keys(TRANSACTIONS.PEGNET_ASSET_LABELS).find(
            key => TRANSACTIONS.PEGNET_ASSET_LABELS[key] === propertyName,
        );
        const keyWithLowerCase = key[0].toLowerCase() + key.slice(1);
        return keyWithLowerCase;
    }
    return propertyName;
};
