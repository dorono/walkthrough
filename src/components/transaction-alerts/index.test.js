import React from 'react';
import {shallow} from 'enzyme';
import {TRANSACTIONS} from 'constants/transactions';
import TransactionAlerts from './index';

describe('TransactionAlerts', () => {
    const defaultProps = {
        transactionType: TRANSACTIONS.STATUSES.PENDING,
    };

    it('should render without errors', () => {
        shallow(<TransactionAlerts {...defaultProps} />);
    });

    it('should match snapshot for a PENDING transaction', () => {
        const defaultProps = {
            transactionType: TRANSACTIONS.STATUSES.PENDING,
        };

        expect(shallow(<TransactionAlerts {...defaultProps} />)).toMatchSnapshot();
    });

    it('should match snapshot for a REJECTED transaction', () => {
        const defaultProps = {
            transactionType: TRANSACTIONS.STATUSES.REJECTED,
        };

        expect(shallow(<TransactionAlerts {...defaultProps} />)).toMatchSnapshot();
    });
});
