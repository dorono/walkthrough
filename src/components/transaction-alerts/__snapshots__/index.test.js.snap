// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransactionAlerts should match snapshot for a PENDING transaction 1`] = `
<div
  className="root transactionAlertLegendHighlight"
>
  <Icon
    className="transactionAlertLegendIcon"
    name="Schedule"
  />
  <span>
    <strong>
      Pending
      : 
    </strong>
     
    This transaction has not yet been processed by PegNet.
  </span>
</div>
`;

exports[`TransactionAlerts should match snapshot for a REJECTED transaction 1`] = `
<div
  className="root transactionAlertLegendHighlight"
>
  <Icon
    className="transactionAlertLegendIcon"
    name="ErrorOutlineIcon"
  />
  <span>
    <strong>
      Rejected
      : 
    </strong>
     
    This transaction has been rejected due to insufficient funds.
  </span>
</div>
`;
